### PR TITLE
Auto update package versions upon release generation

### PR DESCRIPTION
**Pull request by damirault**
- Adapted from .Net package incrementing
  - https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Update-PkgVersion.ps1
- Seed Changelog file to allow updates to happen
- Versioning conventions are draft implementations, full semantics to be defined still at https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#incrementing-after-release-cpp

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-cpp-release.yml [hunk 1]
```diff
@@ -65,3 +65,28 @@ stages:
...
+
+              strategy:
+                runOnce:
+                  deploy:
+                    steps:
+                      - checkout: self
+                      - pwsh: |
+                          eng/Update-PkgVersion.ps1 -ServiceDirectory '${{parameters.ServiceDirectory}}' -PackageName '${{artifact.name}}'
+                        displayName: Increment package version
+                      - template: tools/clone-buildtools/clone-buildtools.yml@azure-sdk-build-tools
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-cpp-release.yml]: We should start using the templates in eng/common for this and other things in this file now that @chidozieononiwu has ported them there.

File: eng/Update-PkgVersion.ps1 [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    Set-Content -Path $PackageVersionPath -Value ${NewPackageVersion}
+
+    # Increment Version in ChangeLog file
+    & "${PSScriptRoot}/common/Update-Change-Log.ps1" -Version $NewPackageVersion -ChangeLogPath $ChangelogPath -Unreleased $Unreleased -ReplaceVersion $ReplaceVersion
+}
+
+# Parse a VersionString to verify that it is right
+function Parse-Version($VerisionString)
+{
+    if ([System.String]::IsNullOrEmpty($VerisionString))
```
Comment (weshaggard) [inline eng/Update-PkgVersion.ps1]: https://github.com/Azure/azure-sdk/blob/6903dad6131be2a81469ae2d4a55f408469ad1ee/eng/scripts/Update-Release-Versions.ps1#L26-L58

That is some code we have that we are using to parse versions. We should probably figure out a way to share that but at the very least we should be consistent with it.

File: eng/Update-PkgVersion.ps1 [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+{
+    Write-Verbose "New Version: ${NewPackageVersion}"
+    Set-Content -Path $PackageVersionPath -Value ${NewPackageVersion}
+
+    # Increment Version in ChangeLog file
+    & "${PSScriptRoot}/common/Update-Change-Log.ps1" -Version $NewPackageVersion -ChangeLogPath $ChangelogPath -Unreleased $Unreleased -ReplaceVersion $ReplaceVersion
+}
+
+# Parse a VersionString to verify that it is right
+function Parse-Version($VerisionString)
```
Comment (weshaggard) [inline eng/Update-PkgVersion.ps1]: typo: VerisionString -> VersionString

File: eng/Update-PkgVersion.ps1 (hunk 2 above)
Comment (weshaggard) [reply eng/Update-PkgVersion.ps1]: Looks like you just copied it from .NET (https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Update-PkgVersion.ps1#L48)... even more reason to refactor this. @chidozieononiwu can you create an issue to create some shared versioning helper scripts.

File: eng/pipelines/templates/stages/archetype-cpp-release.yml (hunk 1 above)
Comment (damirault) [reply eng/pipelines/templates/stages/archetype-cpp-release.yml]: Created #56 to track this across the repo. It looks like Clone-BuildTools itself is _not_ yet in eng/common.

File: eng/Update-PkgVersion.ps1 (hunk 2 above)
Comment (damirault) [reply eng/Update-PkgVersion.ps1]: Created https://github.com/Azure/azure-sdk-tools/issues/469 to track

**Pull request merged**